### PR TITLE
BUG: Issue #4464 Fixed residuez which failed for Complex Numbers

### PR DESCRIPTION
**Pull request by deepakgouda**
Fixed the bug in issue #4464 
The function raised a TypeError:
```
TypeError: Cannot cast ufunc subtract output from dtype('complex128') to dtype('float64') with casting rule 'same_kind'
```
To fix it, I fixed the polydiv function in `numpy/lib/polynomial.py`. I created a PR in [numpy repo](https://github.com/numpy/numpy/pull/10473). 
Further, in cases with repeated roots, the obtained roots were of the form 0.9999998 and 1.000001. These were treated as two different roots with multiplicity 1 each. This led to wrong values of the residue. I rounded them off to 3 decimal digits as the `tol` value was set to be 1e-3. This gives the correct answer.

Previous Commit:

Attempt to fix documentation issue [#7750 ](https://github.com/scipy/scipy/issues/7750)
According to the behavior of the function and the tests given in _scipy/stats/tests/test_mstats_basic.py_ when the inclusive tuple is True, the number of data being masked on each side gets truncated and when its False, it gets rounded. 

I changed the documentation accordingly i.e. 
> inclusive : {(True, True) tuple}, optional
>         Tuple indicating whether the number of data being masked on each side
>         should be truncated (True) or rounded (False).

### CODE REVIEW

File: scipy/signal/signaltools.py [hunk 1]
```diff
@@ -1978,6 +1979,14 @@ def residuez(b, a, tol=1e-3, rtype='avg'):
     b = brev[::-1]
     p = roots(a)
     r = p * 0.0
+    for i in range(len(p)):
+        round_p = p[i]*0.0
+        if iscomplex(p[i]):
+            round_p = round_p+round(p[i].real,3)
+            round_p = round_p+round(p[i].imag,3)*1j
+        else:
+            round_p = round(p[i],3)
```
Comment (eric-wieser) [inline scipy/signal/signaltools.py]: What about when the caller sets `tol=1e-12`?

**Review by eric-wieser — CHANGES_REQUESTED** [review]

**Comment by eric-wieser** [comment]
Deliberately rounding still seems very strange to me

**Comment by deepakgouda** [comment]
In a test case with repeated roots, while one root stood at `9.99999993e-01-1.61620816e-08j`, the other stood at `1.00000001e+00+1.61620814e-08j` and they were treated as two different roots with multiplicity 1 each. I tried the same in MATLAB and it generated the values as `1.0000+0.0000i` and `1.0000+0.0000i`.

Treating the two roots as different, gave wrong values of the residue. So, either we can round off to the tolerance level or we can try to compute accurate values of the root upto the 8th decimal. So, I went with rounding off.

File: scipy/stats/mstats_basic.py [hunk 2]
```diff
@@ -1747,7 +1747,7 @@ def winsorize(a, limits=None, inclusive=(True, True), inplace=False,
         indicate an open interval.
     inclusive : {(True, True) tuple}, optional
         Tuple indicating whether the number of data being masked on each side
-        should be rounded (True) or truncated (False).
```
Comment (eric-wieser) [inline scipy/stats/mstats_basic.py]: You should make this fix in a separate PR

Comment (deepakgouda) [reply scipy/stats/mstats_basic.py]: Sorry for committing this in the same PR(I hadn't created a separate branch so, it got pushed into the same PR). I reverted this commit and will create a separate PR.

Comment (deepakgouda) [reply scipy/stats/mstats_basic.py]: I am closing this PR and will make two separate pull requests.
